dotnet/src/SemanticKernel.Abstractions/AI/TextCompletion/TextCompletionExtensions.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel.AI.TextCompletion;
+
+/// <summary>
+/// This sponsor class Keeps support to the previous methods to avoid any breaking changes.
+/// </summary>
+public static class TextCompletionExtensions
+{
+    /// <summary>
+    /// Creates a completion for the prompt and settings.
+    /// </summary>
+    /// <param name="textCompletion">Target interface to extend</param>
+    /// <param name="text">The prompt to complete.</param>
+    /// <param name="requestSettings">Request settings for the completion API</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>Text generated by the remote model</returns>
+    public static async Task<string> CompleteAsync(this ITextCompletion textCompletion,
+        string text,
+        CompleteRequestSettings requestSettings,
+        CancellationToken cancellationToken = default)
+    {
+        var completions = await textCompletion.GetCompletionsAsync(text, requestSettings, cancellationToken).ConfigureAwait(false);
+
+        if (completions.Count == 0)
+        {
+            throw new AIException(AIException.ErrorCodes.InvalidResponseContent, "No completion result");
+        }
+
+        if (completions.Count > 1)
+        {
+            throw new NotSupportedException($"{nameof(CompleteAsync)} don't support multiple completions results. Use {nameof(ITextCompletion.GetCompletionsAsync)} instead.");
```

Comment:
I don't understand this part: the client returned multiple completions and the error seems to say the opposite. Also why is this an error, when would it happen? Could you add some comment explaining the scenario?

Reply:
If any user attemps to use the extension methods providing resultsperprompt > 1 they will get the error. I can just ignore the other results silently or throw, either options is fine for me, but IMHO hidding/losing info is worse than failing.

Reply:
That's a fair point and I also noted that currently the Results implementation actually loops on all the choices found, I changed the behavior on the non streaming to actually be the same. Not adding error implementation I reduce the changes of breaking any code that were using those methods before.